src/config/validation.rs

```diff
@@ -189,6 +192,137 @@ pub fn check_outputs(config: &ConfigBuilder) -> Result<(), Vec<String>> {
     }
 }
 
+pub async fn check_buffer_preconditions(config: &Config) -> Result<(), Vec<String>> {
+    // We need to assert that Vector's data directory is located on a mountpoint that has enough
+    // capacity to allow all sinks with disk buffers configured to be able to use up to their
+    // maximum configured size without overrunning the total capacity.
+    //
+    // More subtly, we need to make sure we properly map a given buffer's data directory to the
+    // appropriate mountpoint, as it is technically possible that individual buffers could be on
+    // separate mountpoints.
```

Comment:
We don't actually have per sink buffer configuration for the location, right? This would be possible by predetermining how Vector will lay out the buffers and then constructing the underlying filesystem?

Reply:
So, yeah, Vector uses a common global data directory under which all buffers will have their data directory, but think of things like Kubernetes where you often bind mount on top of an existing directory to override something, etc.

That's trivial to do on Linux in general, even if it is likely very uncommon for Vector users to do. So we could have a case where operators use individual storage devices, mounted such that they specifically take over for `/globaldatadir/buffer/v2/sink_id`, which is why we go to the lengths above to make sure we figure out the precise mountpoint each buffer's data directory is _actually_ located on.

Reply:
👍 I was thinking through, as well, if allowing per-buffer configuration of that directory is desirable. We do have per-component configuration for where checkpoints get stored (at least for some components)

Reply:
Yeah, not sure. It would certainly make the above scenario easier for operators if they did in fact want to specifically stick buffer data elsewhere.